robottelo/report_portal/portal.py

```diff
@@ -226,45 +168,71 @@ def _test_params(self, status, defect_type, user):
-
-        :returns tuple (int, list): Total pages count and the list of tests along with
-            each tests properties in a page
-        """
-        params['page.page'] = page
-        resp = requests.get(
-            url=f'{self.report_portal.api_url}/item',
-            headers=self.report_portal.headers,
-            params=params,
-            verify=False,
-        )
-        resp.raise_for_status()
-        total_pages = resp.json()['page']['totalPages']
-        pagedata = resp.json().get('content')
-        return total_pages, pagedata
-
-    def tests(self, status=None, defect_type=None, user=None):
+    def get_tests(self, launch=None, test_args=None):
         """Returns tests data customized by kwargs parameters.
 
         This is a main function that will be called to retrieve the tests data
         of a particular test status or/and defect_type
 
+        :param str launch: Dict of a target launch to fetch test items for
         :param str status: Filter tests of a launch with tests `status`
         :param str defect_type: Filter tests of a launch with tests `defect_type`
         :returns dict: All filtered tests dict based on params data keyed by test name and test
             properties as value, in format -
             ```{'test_name1':test1_properties_dict, 'test_name2':test2_properties_dict}```
         """
+        params = {
+            'page.size': 50,
+            'page.sort': 'name',
+            'filter.eq.launchId': launch["id"],
+            'filter.ne.type': "SUITE",
+        }
+
+        # parse the test filter parameters and turn them into API filter parameters
+        if test_args is None:
+            test_args = {}
+        if not isinstance(test_args, dict):
+            raise TypeError('test_args is expected to be of type dict')
+        if test_args.get('status'):
+            params['filter.in.status'] = ','.join(test_args['status']).upper()
+        if test_args.get('defect_types'):
+            params['filter.in.statistics.defects'] = test_args['defect_types']
+
+        # send HTTP request to RP API, retrieve the paginated results and join them together
         page = 1
-        params = self._test_params(status, defect_type, user)
-        total_pages, data = self._test_requester(params=params, page=page)
-        while page < total_pages:
+        total_pages = 1
+        resp_tests = []
+        while page <= total_pages:
+            logger.debug(page)
+            params['page.page'] = page
+            resp = requests.get(
+                url=f'{self.api_url}/item',
+                headers=self.headers,
+                params=params,
+                verify=False,
+            )
+            resp.raise_for_status()
+            resp_tests.extend(resp.json()['content'])
+            total_pages = resp.json()['page']['totalPages']
+            page += 1
+
+        # Only select tests matching the supplied paths
```

Comment:
This section should not be really necessary (the final de-select process should take care of unfiltered results too), but I wanted to keep it here as a placeholder for a case, the RP search bug gets resolved, so we can turn this into the API filter parameters and move the heavy lifting to the server side.